### DISCUSSION TITLE
Can't add NavigationItem's to topNavigation?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9004] [Help]
I'd like to utilize both the default sidebar **navigation** and **topNavigation**. Unfortunately, I'm unable to add items to **topNavigation**.

```
use App\Filament\Pages\Settings;
use App\Filament\Resources\UserResource;
use Filament\Navigation\NavigationBuilder;
use Filament\Navigation\NavigationItem;
use Filament\Pages\Example;
use Filament\Panel;
 
public function panel(Panel $panel): Panel
{
    return $panel
        // ...
        ->topNavigation(function (NavigationBuilder $builder): NavigationBuilder {
            return $builder->items([
                NavigationItem::make('Example')
                    ->icon('heroicon-o-home')
                    ->isActiveWhen(fn (): bool => request()->routeIs('#'))
                    ->url(fn (): string => Example::getUrl()),
            ]);
        });
}
```

**Comment by leandrocfe** [comment]
You can add navigation items using
```php
$panel
->topNavigation()
->navigationItems([
    NavigationItem::make('Custom Item')
        ->url('#')
        ->icon('heroicon-o-presentation-chart-line')
])
```

**Reply by devhoussam1998** [reply]
Maybe you dont understand what I want to achieve read first the problem not adding a **NavigationItem** the problem is I can't add  **NavigationItem** in -`>topNavigation()` I'm trying to use both sidebar navigation (the default navigation) and top navigation.

**Reply by leandrocfe** [reply]
Isn't possible I think, but you can publish the views
For example
original file: vendor/filament/filament/resources/views/components/topbar/index.blade.php
You can create a custom view in resources/views/vendor/filament-panels/components/topbar/index.php

and edit what you want
